document.go

```diff
@@ -37,6 +47,17 @@ func WithSegmentation(include bool) DocOpt {
 	}
 }
 
+// WithConcurrency can enable (the default) or disable sentence segmentation.
+func WithConcurrency(include bool) DocOpt {
```

Comment:
minor nit: I am not a fan of sending bool parameters in functions

Suggestion: 

```
// something like this
func WithConcurrency() DocOpt {
	return func(doc *Document, opts *DocOpts) {
		maxWorkers := runtime.NumCPU()
		opts.Concurrency = true
		opts.wp = workerpool.New(maxWorkers)
	}
}
```